trigger/task.go

```diff
@@ -635,37 +635,56 @@ func (t *taskTriggers) shouldIncludeTest(sub *event.Subscription, previousTask *
 	if test.Status != evergreen.TestFailedStatus {
 		return false, nil
 	}
-	record, err := alertrecord.FindByLastTaskRegressionByTest(sub.ID, test.TestFile, t.task.DisplayName, t.task.BuildVariant, t.task.Project)
+
+	alertForTask, err := alertrecord.FindByTaskRegressionByTaskTest(sub.ID, test.TestFile, t.task.DisplayName, t.task.BuildVariant, t.task.Project, t.task.Id)
 	if err != nil {
-		return false, errors.Wrap(err, "Failed to fetch alert record")
+		return false, errors.Wrap(err, "can't find alerts for task test")
+	}
+	// we've already alerted for this task
+	if alertForTask != nil {
+		return false, nil
 	}
-	if record != nil {
-		if record.RevisionOrderNumber == t.task.RevisionOrderNumber || previousTask == nil {
-			return false, nil
-		}
 
-		oldTestResult, ok := t.oldTestResults[test.TestFile]
-		if !ok {
-			if test.Status != evergreen.TestFailedStatus {
-				return false, nil
-			}
+	// a test in a new task is defined as a regression
+	if previousTask == nil {
+		return true, nil
+	}
 
-		} else if !isTestStatusRegression(oldTestResult.Status, test.Status) {
-			if len(record.TaskId) != 0 {
-				var isOld bool
-				isOld, err = taskFinishedTwoOrMoreDaysAgo(record.TaskId, sub)
-				if err != nil || !isOld {
-					return false, errors.Wrap(err, "failed to fetch last alert age")
-				}
-			}
-		}
+	oldTestResult, ok := t.oldTestResults[test.TestFile]
+	// a new test in an existing task is defined as a regression
+	if !ok {
+		return true, nil
```

Comment:
Move these to the top so that we can short circuit the query

Reply:
done.

Reply:
on second thought: the query needs to be first: if we've already alerted for this task (like if it's been restarted) we don't want to alert again even if the task or test are new.